tests/di/container_cache_directory_test.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace
+{
+	class container_cache_directory_test extends \phpbb_test_case //phpbb_di_container_test
+	{
+		protected $config_php;
+
+		/**
+		* @var \phpbb\di\container_builder
+		*/
+		protected $builder;
+		protected $phpbb_root_path;
+		protected $filename;
+
+		public function setUp(): void
+		{
+			$this->phpbb_root_path = dirname(__FILE__) . '/';
+			$this->config_php = new \phpbb\config_php_file($this->phpbb_root_path . 'fixtures/', 'php');
+			$this->builder = new phpbb_mock_phpbb_di_container_builder($this->phpbb_root_path . 'fixtures/', 'php');
+			$this->builder->with_config($this->config_php);
+
+			$this->filename = $this->phpbb_root_path . '../tmp/container.php';
+			if (is_file($this->filename))
+			{
+				unlink($this->filename);
+			}
+
+			parent::setUp();
+		}
+
+
+		public function test_cache_directory_can_be_overridden()
+		{
+			$newCacheDirectory = $this->phpbb_root_path . "fixtures/overwrite-cache-directory/test/";
```

Comment:
Please don't use camelCase in PHP files (see also our coding guidelines). We're also using single quotes.
```suggestion
			$new_cache_directory = $this->phpbb_root_path . 'fixtures/overwrite-cache-directory/test/';
```